src/util/protected-route-wrapper/ProtectedRouteWrapper.tsx

```diff
@@ -0,0 +1,36 @@
+import { Fragment, ReactNode, useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { validateUserToken } from './service/protected-route-wrapper-service';
+
+interface Props {
+  children: ReactNode | ReactNode[]
+}
+
+const ProtectedRouteWrapper = (props: Props) => {
+  const navigate = useNavigate();
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+
+  const checkUserToken = () => {
+    const userToken = localStorage.getItem('user-token');
+
+    if (!validateUserToken(userToken)) {
+      setIsLoggedIn(false);
+      return navigate('/login');
+    }    
+    setIsLoggedIn(true);
+  };
+
+  useEffect(() => {
+    checkUserToken();
+  }, [isLoggedIn]);
+
+  return (
+    <Fragment>
+      {
+        isLoggedIn ? props.children : null
+      }
+    </Fragment>
```

Comment:
Security concern: children for sure won't be visible, but will it be processed? E.g. children component can create a secret cookie, which will be generated before redirecting to `/login`.

Reply:
`isLoggedIn` is false by default and if the token check returns that it's invalid, the `props.children` are never rendered, so I don't think that your scenario is possible

Reply:
What I mean is that it's not like:

`isLoggedIn` is true --> `props.children` are rendered --> token validation returns false --> `isLoggedIn := false` --> redirect to `/login`

Because it's without first two steps

Reply:
Sanity check passed.